### PR TITLE
feat: put airgap client build in a separate folder

### LINKED ISSUE #22414: [Task]: Separate build output directory for airgap client builds
### Is there an existing issue for this?

- [X] I have searched the existing issues

### SubTasks

Now that the CI will be running client build for both EE and Airgap, we need a separate build output dir to build docker image from for airgapped builds.

### PR DESCRIPTION
**Pull request by berzerkeer**
## Description

- Adds a new build output dir `build_airgap` for airgap client builds. 

Fixes #22414 

Media
> A video or a GIF is preferred. when using Loom, don’t embed because it looks like it’s a GIF. instead, just link to the video

## Type of change

- New feature (non-breaking change which adds functionality)
- Chore (housekeeping or task changes that don't impact user perception)

## How Has This Been Tested?

- Manual

### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: app/client/build.sh [hunk 1]
```diff
@@ -6,13 +6,21 @@ GIT_SHA=$(eval git rev-parse HEAD)
 echo $GIT_SHA
 echo "Sentry Auth Token: $SENTRY_AUTH_TOKEN"
 
+if [ "$REACT_APP_AIRGAP_ENABLED" = "true" ]; then
+    echo "Building for airgap"
+    OUTPUT_PATH=build_airgap
+else
+    echo "Building for regular"
```
Comment (riodeuno) [inline app/client/build.sh:13]: The term `regular` can be misleading, let's call it `Building for non-airgapped Appsmith instances` and `Building for airgapped Appsmith intances`

**Review by riodeuno — COMMENTED** [review]
Once comment. The rest looks good to me.

File: app/client/build.sh (hunk 1 above)
Comment (berzerkeer) [reply app/client/build.sh]: Done

**Comment by berzerkeer** [comment]
> Once comment. The rest looks good to me.

I have updated the PR, there was also one fix where I was treating the env variables as Boolean but those are strings.

**Review by yaldram — APPROVED** [review]

**Review by sharat87 — APPROVED** [review]

**Pull request merged**